sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -449,14 +449,18 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       ResolveEncodersInUDF),
     // The rewrite rules might move resolved query plan into subquery. Once the resolved plan
     // contains ScalaUDF, their encoders won't be resolved if `ResolveEncodersInUDF` is not
-    // applied before the rewrite rules. So we need to apply `ResolveEncodersInUDF` before the
-    // rewrite rules.
+    // applied before the rewrite rules. So we need to apply the rewrite rules after
+    // `ResolveEncodersInUDF`
     Batch("DML rewrite", fixedPoint,
```

Comment:
Context of why this batch is not in the main resolution batch: https://github.com/apache/spark/commit/fab6d83ac70f61925edb6eb0b1fb600fb59c9761